app/src/main/AndroidManifest.xml

```diff
@@ -34,6 +34,7 @@
             android:name=".messages.MessagesActivity"
             android:exported="false"
             android:launchMode="singleTask"
+            android:taskAffinity=".messages.MessagesActivity"
```

Comment:
Why is thin needed?

Reply:
According to the instructions in [Android Developer: excludeFromRecent](https://developer.android.com/guide/topics/manifest/activity-element#exclude), if excludeFromRecent wants to take effect, the new activity must be created in a new task, so taskAffinity is added.

Reply:
So we could add the `FLAG_ACTIVITY_NEW_TASK` to the MessageActivity intent, to get the same behavior?

Reply:
But according to my test, FLAG_ACTIVITY_NEW_TASK doesn't really work here